src/main/java/edu/kit/compiler/transform/Lower.java

```diff
@@ -1,12 +1,38 @@
 package edu.kit.compiler.transform;
 
-import firm.Backend;
-import firm.Ident;
-import firm.Program;
-import firm.Util;
+import com.sun.jna.Platform;
+import firm.*;
+import firm.nodes.Node;
 
 public class Lower {
 
+    private static Node calloc;
+
+    public static String makeLdIdent(String str) {
+        if (Platform.isMac() || Platform.isWindows()) {
+            str = "_" + str;
+        }
+        return str;
+    }
+
+    /**
+     * returns the address of calloc in this construction
+     * @param construction
+     * @return
+     */
+    public static Node getCallocNode(Construction construction) {
+        SegmentType global = Program.getGlobalType();
+        if (calloc == null) {
+            MethodType callocType = new MethodType(new Type[]{ TypeMapper.getIntegerType(), TypeMapper.getIntegerType()},
+                    new Type[]{ TypeMapper.getPointerType()});
+
+            Entity callocEntity = new Entity(global, "calloc", callocType);
```

Comment:
You probably should call `Ident.mangleGlobal("calloc")`. (At least we do this for the standard library methods.)